src/Arcus.WebApi.Tests.Unit/Security/Authorization/JwtTokenReaderTests.cs

```diff
@@ -77,5 +78,33 @@ public void Constructor_WithTokenValidationParametersAndDiscoveryEndpoint_Succee
             // Assert
             Assert.NotNull(jwtTokenReader);
         }
+
+        [Fact]
+        public void Constructor_WithClaimCheck_Succeeds()
+        {
+            // Arrange
+            var claimCheck = new ClaimCheck
+            {
+                {"aud", "368e1166-2235-48c7-8685-48d1fc24a9de"},
+                {"oid", "fa323e12-e4b8-4e22-bb2a-b18cb4b76301"},
+                {"aad", "ecc1e1c0-61bc-4e87-bd41-02333aa485bd"}
+            };
+
+            // Act
+            var jwtTokenReader = new JwtTokenReader(claimCheck);
+
+            // Assert
+            Assert.NotNull(jwtTokenReader);
+        }
+
+        [Fact]
+        public void Constructor_WithoutClaimCheck_ThrowsException()
+        {
+            // Arrange
+            var claimCheck = new ClaimCheck();
+
+            // Act & Assert
+            Assert.Throws<ArgumentException>(() => new JwtTokenReader(claimCheck));
+        }
     }
```

Comment:
It would be nice to have some tests which check if the functionality regarding the claim checking works.

Reply:
done